### PR TITLE
[DependencyInjection] anonymous services are always private

### PR DESCRIPTION
**Pull request by xabbuh**
| Q | A |
| --- | --- |
| Branch? | 2.7 |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets |  |
| License | MIT |
| Doc PR |  |

### CODE REVIEW

File: src/Symfony/Component/DependencyInjection/Loader/XmlFileLoader.php [hunk 1]
```diff
@@ -302,11 +302,7 @@ private function processAnonymousServices(\DOMDocument $xml, $file)
                 // give it a unique name
                 $id = sprintf('%s_%d', hash('sha256', $file), ++$count);
                 $node->setAttribute('id', $id);
-
-                if ($services = $this->getChildren($node, 'service')) {
```
Comment (xabbuh) [inline src/Symfony/Component/DependencyInjection/Loader/XmlFileLoader.php]: This doesn't seem like it made much sense (was introduced in #10619).

**Comment by HeahDude** [comment]
Thanks! It's been an hour since I'm trying to understand what's wrong with this :)

This change you mention might be useless after #18246.

**Comment by HeahDude** [comment]
Sorry wrong link :)

**Comment by nicolas-grekas** [comment]
:+1:

**Comment by nicolas-grekas** [comment]
Thank you @xabbuh.

**Comment by DHager** [comment]
Reporting a BC break with #18703, although perhaps you could argue that our application had a long-standing bug that was simply revealed. :confused:

**Pull request merged**